File: sdk/python/examples/stream_chat.py
"""
Tortoise Python SDK - Streaming Chat Example
"""

import asyncio
from tortoise import TortoiseClient


async def main():
    async with TortoiseClient(
        gateway_url="http://localhost:8080",
        api_key="your-api-key"
    ) as client:
        print("Starting streaming chat...\n")
        
        # Stream response
        full_response = ""
        async for chunk in client.chat_stream(
            message="Write a short poem about artificial intelligence.",
            temperature=0.8
        ):
            print(chunk, end="", flush=True)
            full_response += chunk
        
        print("\n\n[Streaming complete]")


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/rust/examples/stream_chat.rs
//! Tortoise Rust SDK - Streaming Chat Example

use futures_util::StreamExt;
use tortoise_sdk::{Client, ClientConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let config = ClientConfig::default()
        .with_url("http://localhost:8080")
        .with_api_key("your-api-key");
    
    let client = Client::new(config);
    
    println!("Starting streaming chat...\n");
    
    // Create streaming request
    let stream = client.chat_stream("Write a haiku about Rust programming.").await?;
    
    // Process stream
    let mut stream = stream?;
    while let Some(chunk_result) = stream.next().await {
        match chunk_result {
            Ok(text) => print!("{}", text),
            Err(e) => eprintln!("Error: {}", e),
        }
    }
    
    println!("\n\n[Streaming complete]");
    Ok(())
}
